core/network/types/block_request.hpp

```diff
@@ -63,6 +128,44 @@ namespace kagome::network {
     /// used when unspecified
     boost::optional<uint32_t> max;
   };
+
+  /**
+   * @brief compares two BlockRequest instances
+   * @param lhs first instance
+   * @param rhs second instance
+   * @return true if equal false otherwise
+   */
+  inline bool operator==(const BlockRequest &lhs, const BlockRequest &rhs) {
+    return lhs.id == rhs.id && lhs.fields == rhs.fields && lhs.from == rhs.from
+           && lhs.to == rhs.to && lhs.direction == rhs.direction
+           && lhs.max == rhs.max;
+  }
+
+  /**
+   * @brief outputs object of type BlockRequest to stream
+   * @tparam Stream output stream type
+   * @param s stream reference
+   * @param v value to output
+   * @return reference to stream
+   */
+  template <class Stream,
+            typename = std::enable_if_t<Stream::is_encoder_stream>>
+  Stream &operator<<(Stream &s, const BlockRequest &v) {
+    return s << v.id << v.fields << v.from << v.to << v.direction << v.max;
+  }
+
+  /**
+   * @brief decodes object of type BlockRequest from stream
+   * @tparam Stream input stream type
+   * @param s stream reference
+   * @param v value to decode
+   * @return reference to stream
+   */
+  template <class Stream,
+            typename = std::enable_if_t<Stream::is_decoder_stream>>
+  Stream &operator>>(Stream &s, BlockRequest &v) {
+    return s >> v.id >> v.fields >> v.from >> v.to >> v.direction >> v.max;
```

Comment:
Have you checked the order of encoding/decoding with some another implementation? Just asking, because if you haven't, you should

Reply:
The order of fields corresponds to the order in rust implementation. The order of encryption is not defined explicitly and is the same as the order of fields, so it's ok.